### PR TITLE
Math tools to create confidence intervals from data

### PR DESCRIPTION
**Pull request by tergol1**
#### What is the purpose of this pull request?
This is the first part of two that should add the capability of calculate confidence intervals for certain metrics, given the data got during a AB test.

#### How should this be manually tested?
As this don't add any feature that affect any API, the bets way to test it is by linking this app with this change and also a log of the calculated value by these function added in this PR. Then compare the returned value with the expected.

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

**Review by valentino-amadeus — COMMENTED** [review]
.

File: node/utils/mathTools/statistics/normal.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    if (p <= P_HIGH)
+    {
+        q = p - 0.5
+        const r = q * q
+        return (((((a[0] * r + a[1]) * r +a[2]) * r + a[3]) * r + a[4]) * r +a[5]) * q / (((((b[0] * r + b[1]) * r + b[2]) * r + b[3]) * r + b[4]) * r + 1)
+    }
+
+    return -(((((c[0] * q + c[1]) * q + c[2]) * q + c[3]) * q + c[4]) * q + c[5]) / ((((d[0] * q + d[1]) * q + d[2]) * q + d[3]) * q +1)
+
+}
```
[the remark this answers is not in the document]
Comment (valentino-amadeus) [reply node/utils/mathTools/statistics/normal.ts:60]: And... We actually don't need to implement this 🙈 . [There's a lib for it](https://www.npmjs.com/package/gaussian).

File: node/utils/mathTools/statistics/t-student.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+        y = y > 0.002 ? Math.exp(y) - 1 : 0.5 * y * y + y
+    }
+    else
+    {
+        y =  ((1.0/(((n+6.0)/(n * y)-0.089 * d - 0.822) * (n + 2.0) * 3.0)+0.5/(n + 4.0)) * y - 1.0) * (n+1.0)/(n+2.0) + 1.0/y
+    }
+
+    return Math.sqrt(n * y)
+}
```
Comment (valentino-amadeus) [inline node/utils/mathTools/statistics/t-student.ts:55]: There's a lib for this as well: https://www.npmjs.com/package/js-stats

File: node/utils/mathTools/statistics/confidenceInterval.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { inverseTStudent } from './t-student'
+
+export interface ConfidenceInterval
+{
```
Comment (valentino-amadeus) [inline node/utils/mathTools/statistics/confidenceInterval.ts:4]:
```suggestion
interface ConfidenceInterval {

```

File: node/utils/mathTools/statistics/confidenceInterval.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    LowerBound: number
+    UpperBound: number
+}
+
+export function confidenceInterval(degreesOfFreedom: number, confidence: number, mean: number, stdDev: number): ConfidenceInterval
+{
```
Comment (valentino-amadeus) [inline node/utils/mathTools/statistics/confidenceInterval.ts:10]:
```suggestion
export function confidenceInterval(degreesOfFreedom: number, confidence: number, mean: number, stdDev: number): ConfidenceInterval {

```

**Review by valentino-amadeus — COMMENTED** [review]
LGTM.

**Review by valentino-amadeus — APPROVED** [review]
LGTM.

File: node/utils/mathTools/statistics/normal.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (tergol1) [reply node/utils/mathTools/statistics/normal.ts:60]: Did you checked the lib's algorithm? I think that one is better (or maybe just better written as I looked at the one in the lib's repository and didn't liked it)

[the remark this answers is not in the document]
Comment (valentino-amadeus) [reply node/utils/mathTools/statistics/normal.ts:60]: In the lib, they tried to implement an inverse to the complement of the [error function](https://en.wikipedia.org/wiki/Error_function). I hardly believe that that's better than what we have. Your point on the code's visibility is in fact pertinent, so I think we can stick with what we already have.

**Pull request merged**